beacon-chain/core/helpers/validators.go

```diff
@@ -202,7 +251,7 @@ func BeaconProposerIndex(state *stateTrie.BeaconState) (uint64, error) {
 //        if effective_balance * MAX_RANDOM_BYTE >= MAX_EFFECTIVE_BALANCE * random_byte:
 //            return ValidatorIndex(candidate_index)
 //        i += 1
-func ComputeProposerIndex(validators []*ethpb.Validator, activeIndices []uint64, seed [32]byte) (uint64, error) {
+func ComputeProposerIndexWithValidators(validators []*ethpb.Validator, activeIndices []uint64, seed [32]byte) (uint64, error) {
```

Comment:
Do we still need this? Can it be marked as deprecated?

```suggestion
//
// Deprecated: Prefer using the beacon state with ComputeProposerIndex to avoid an unnecessary copy of the validator set.
func ComputeProposerIndexWithValidators(validators []*ethpb.Validator, activeIndices []uint64, seed [32]byte) (uint64, error) {
```

If deprecated is not appropriate, please still put a message somewhere that users should prefer the other method whenever possible.Otherwise we could still see misuse with `ComputeProposerIndexWithValidators(state.Validators(), ...)`

Reply:
I think deprecated would be appropriate here. This is only still there because an archival method requires this, but that method is going to be deprecated anyway